go.mod: upgrade lib/pq to v1.10.2

otan opened the pull request:
Resolves #65066.

See individual commits for details.

otan committed:
go.mod: upgrade lib/pq to v1.10.2

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/65319)
<!-- Reviewable:end -->

otan commented:
panic seems to be caused by https://github.com/lib/pq/pull/1004.

otan commented:
given that the test works on PostgreSQL, it seems that CockroachDB would be the issue here...
specifically when querying and having 0 results from the query.

otan committed:
logictest/timetz: add regression test for second offset TimeTZ

Release note (bug fix): Fixed a bug where TimeTZ values would not
display appopriately in the CLI.

otan commented:
abandoning for now
